main.tf

```diff
@@ -159,9 +159,9 @@ resource "google_alloydb_instance" "primary" {
     content {
       enable_public_ip = var.primary_instance.enable_public_ip
       dynamic "authorized_external_networks" {
-        for_each = var.primary_instance.cidr_range == null ? [] : ["authorized_external_networks"]
+        for_each = var.primary_instance.cidr_range != null ? var.primary_instance.cidr_range : []
```

Comment:
@rayanebel whats the point of changing this? It is working fine and new code is not changing anything except it will show diff in existing user's resource? Can you plz revert this change?

Reply:
@imrannayer The current code is not working when you add a network CIDR. Here the result I have If I revert the code

```bash
 Error: Error creating Instance: googleapi: Error 400: The request was invalid: invalid authorized network CIDR range "authorized_external_networks"
│ Details:
│ [
│   {
│     "@type": "type.googleapis.com/google.rpc.BadRequest",
│     "fieldViolations": [
│       {
│         "field": "networkConfig.authorizedExternalNetworks"
│       }
│     ]
│   },
│   {
│     "@type": "type.googleapis.com/google.rpc.RequestInfo",
│     "requestId": "dab977d705ea321c"
│   }
│ ]
│
│   with module.alloy-db.google_alloydb_instance.primary,
│   on .terraform/modules/alloy-db/main.tf line 146, in resource "google_alloydb_instance" "primary":
│  146: resource "google_alloydb_instance" "primary" {
│
╵
```

**Terraform plan**
```bash
 + machine_config           = [
          + {
              + cpu_count = 2
            },
        ]
      + name                     = (known after apply)
      + network_config           = [
          + {
              + authorized_external_networks = [
                  + {
                      + cidr_range = "authorized_external_networks"
                    },
                ]
              + enable_public_ip             = true
            },
        ]
      + psc_instance_config      = []
```

Reply:
thats cos you have changed line `164`. If you keep line `162` and 164 same, code will work fine. Just revert all the changes you made in `network_config` block.

```
  dynamic "network_config" {
    for_each = var.primary_instance.enable_public_ip ? ["network_config"] : []
    content {
      enable_public_ip = var.primary_instance.enable_public_ip
      dynamic "authorized_external_networks" {
        for_each = var.primary_instance.cidr_range == null ? [] : ["authorized_external_networks"]
        content {
          cidr_range = var.primary_instance.cidr_range
        }
      }
    }
  }
```